src/test/java/io/confluent/connect/jdbc/JdbcSourceTaskUpdateTest.java

```diff
@@ -128,6 +128,29 @@ public void testManualIncrementing() throws Exception {
   }
 
   @Test
+  public void testPrimaryKeyIncrementing() throws Exception {
+    expectInitializeNoOffsets(Arrays.asList(SINGLE_TABLE_PARTITION));
+
+    PowerMock.replayAll();
+
+    db.createTable(SINGLE_TABLE_NAME,
+            "id", "INT NOT NULL PRIMARY KEY");
+    db.insert(SINGLE_TABLE_NAME, "id", 1);
+
+    startTask(null, null, "true", null, 0L);
+
+    verifyIncrementingKeyFirstPoll(TOPIC_PREFIX + SINGLE_TABLE_NAME);
+
+    // Adding records should result in only those records during the next poll()
+    db.insert(SINGLE_TABLE_NAME, "id", 2);
+    db.insert(SINGLE_TABLE_NAME, "id", 3);
+
+    verifyPoll(2, "id", Arrays.asList(2, 3), false, true, TOPIC_PREFIX + SINGLE_TABLE_NAME);
+
+    PowerMock.verifyAll();
+  }
+
```

Comment:
Might be more relevant if exceptions are thrown due to missing primary keys, but should there be any failure tests?